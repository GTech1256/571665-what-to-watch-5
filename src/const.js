export const FIRST_FILTER_NAME = `All Genres`;

export const FILMS_SHOWED_PER_STEP_COUNT = 8;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const APIRoute = {
  FILMS: `/films`,
  PROMO_FILM: `/films/promo`,
  LOGIN: `/login`,
  COMMENTS: `/comments`
};

export const EMPTY_STATE_VALUE = null;
