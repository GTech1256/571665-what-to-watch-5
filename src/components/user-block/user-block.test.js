import React from "react";
import {MemoryRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {AuthorizationStatus} from "../../const";
import UserBlock from "./user-block";

describe(`UserBlock is rendered correctly`, () => {
  it(`Authed`, () => {
    const tree = renderer
      .create(
          <UserBlock
            authorizationStatus={AuthorizationStatus.AUTH}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Unauthed`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <UserBlock
              authorizationStatus={AuthorizationStatus.NO_AUTH}
            />
          </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
