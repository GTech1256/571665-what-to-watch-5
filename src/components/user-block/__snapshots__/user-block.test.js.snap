// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock is rendered correctly Authed 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="img/avatar.jpg"
      width="63"
    />
  </div>
</div>
`;

exports[`UserBlock is rendered correctly Unauthed 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
